libs/plugins/flow-client/src/lib/core/models/flow/insert-node.ts

```diff
@@ -1,28 +1,62 @@
-import { FlowGraph, GraphNode } from '@flogo-web/lib-client/core';
+import { FlowGraph, GraphNode, NodeType } from '@flogo-web/lib-client/core';
+import { BRANCH_PREFIX } from '@flogo-web/lib-client/diagram';
 
 export function insertNode(
   flowGraph: FlowGraph,
   node: GraphNode,
-  parentId?: string
+  parentId?: string,
+  insertBetween?: boolean
 ): FlowGraph {
-  const parent = flowGraph.nodes[parentId];
+  let parent = flowGraph.nodes[parentId];
   let nodes = flowGraph.nodes;
+  let rootId = flowGraph.rootId;
   if (parent) {
+    if (insertBetween) {
```

Comment:
do we really need this flag? Since a task can only have 1 non-branch task at the most most, can we assume that if the parent node has "non-branch" children then it is an "insert between" operation?

For example:

```ts
if (parent) {
  const { nonBranches, branches } = categorizeChildren(parent);
  const isInBetween = nonBranches.length > 0;
}
```